extensions/schema-compare/src/schemaCompareMainWindow.ts

```diff
@@ -386,16 +387,63 @@ export class SchemaCompareMainWindow {
 		this.tablelistenersToDispose.push(this.differencesTable.onCellAction(async (rowState) => {
 			let checkboxState = <azdata.ICheckboxCellActionEventArgs>rowState;
 			if (checkboxState) {
+				this.differencesTable.checked = undefined;
+				this.differencesTable.selectedRows = [checkboxState.row];
 				let diff = this.comparisonResult.differences[checkboxState.row];
-				await service.schemaCompareIncludeExcludeNode(this.comparisonResult.operationId, diff, checkboxState.checked, azdata.TaskExecutionMode.execute);
-				this.saveExcludeState(checkboxState);
+				const result = await service.schemaCompareIncludeExcludeNode(this.comparisonResult.operationId, diff, checkboxState.checked, azdata.TaskExecutionMode.execute);
+				if (result.success) {
+					this.saveExcludeState(checkboxState);
+
+					// depencies could have been included or excluded as a result, so save their exclude states
+					result.affectedDependencies.forEach(difference => {
+						// find the row of the difference and set it's checkbox
+						const row = this.findDifferenceRow(difference);
```

Comment:
hmm... is it possible to keep a Map <DiffEntry_string, row> where we create the diff entry string with say source_name_target_change_type? Is that possible or has some issues?

I think that will be faster - can take more memory but since it would be just string, not that much I assume.

Reply:
Makes sense. Added a map where the key is sourcename_target_name_updateAction_objectType